[2024-10-31T21:26:57.142+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2024-10-31T21:26:57.219+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: sensor_data_consumer.sensor_data_consumer manual__2024-10-31T21:26:53.004040+00:00 [queued]>
[2024-10-31T21:26:57.236+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: sensor_data_consumer.sensor_data_consumer manual__2024-10-31T21:26:53.004040+00:00 [queued]>
[2024-10-31T21:26:57.237+0000] {taskinstance.py:2306} INFO - Starting attempt 1 of 1
[2024-10-31T21:26:57.264+0000] {taskinstance.py:2330} INFO - Executing <Task(SparkSubmitOperator): sensor_data_consumer> on 2024-10-31 21:26:53.004040+00:00
[2024-10-31T21:26:57.277+0000] {standard_task_runner.py:64} INFO - Started process 446 to run task
[2024-10-31T21:26:57.284+0000] {standard_task_runner.py:90} INFO - Running: ['***', 'tasks', 'run', 'sensor_data_consumer', 'sensor_data_consumer', 'manual__2024-10-31T21:26:53.004040+00:00', '--job-id', '12', '--raw', '--subdir', 'DAGS_FOLDER/***_consumer.py', '--cfg-path', '/tmp/tmpaumu7qdy']
[2024-10-31T21:26:57.288+0000] {standard_task_runner.py:91} INFO - Job 12: Subtask sensor_data_consumer
[2024-10-31T21:26:57.401+0000] {task_command.py:426} INFO - Running <TaskInstance: sensor_data_consumer.sensor_data_consumer manual__2024-10-31T21:26:53.004040+00:00 [running]> on host 0f30954d6a44
[2024-10-31T21:26:57.592+0000] {taskinstance.py:2648} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='Bich Ly' AIRFLOW_CTX_DAG_ID='sensor_data_consumer' AIRFLOW_CTX_TASK_ID='sensor_data_consumer' AIRFLOW_CTX_EXECUTION_DATE='2024-10-31T21:26:53.004040+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2024-10-31T21:26:53.004040+00:00'
[2024-10-31T21:26:57.594+0000] {taskinstance.py:430} INFO - ::endgroup::
[2024-10-31T21:26:57.643+0000] {base.py:84} INFO - Using connection ID 'spark_default' for task execution.
[2024-10-31T21:26:57.644+0000] {spark_submit.py:344} INFO - Spark-Submit cmd: spark-submit --master spark://spark-master:7077 --name KafkaSparkHDFS /opt/bitnami/spark/jobs/spark_streaming_job.py
[2024-10-31T21:27:02.690+0000] {spark_submit.py:495} INFO - python3: can't open file '/opt/bitnami/spark/jobs/spark_streaming_job.py': [Errno 2] No such file or directory
[2024-10-31T21:27:02.733+0000] {spark_submit.py:495} INFO - 24/10/31 21:27:02 INFO ShutdownHookManager: Shutdown hook called
[2024-10-31T21:27:02.744+0000] {spark_submit.py:495} INFO - 24/10/31 21:27:02 INFO ShutdownHookManager: Deleting directory /tmp/spark-a2775b3b-d759-4875-a475-0e3a98526cb7
[2024-10-31T21:27:02.916+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2024-10-31T21:27:02.937+0000] {taskinstance.py:2905} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/models/taskinstance.py", line 465, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/models/taskinstance.py", line 432, in _execute_callable
    return execute_callable(context=context, **execute_callable_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/models/baseoperator.py", line 401, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 157, in execute
    self._hook.submit(self._application)
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 426, in submit
    raise AirflowException(
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master spark://spark-master:7077 --name KafkaSparkHDFS /opt/bitnami/spark/jobs/spark_streaming_job.py. Error code is: 2.
[2024-10-31T21:27:02.946+0000] {taskinstance.py:1206} INFO - Marking task as FAILED. dag_id=sensor_data_consumer, task_id=sensor_data_consumer, run_id=manual__2024-10-31T21:26:53.004040+00:00, execution_date=20241031T212653, start_date=20241031T212657, end_date=20241031T212702
[2024-10-31T21:27:02.973+0000] {standard_task_runner.py:110} ERROR - Failed to execute job 12 for task sensor_data_consumer (Cannot execute: spark-submit --master spark://spark-master:7077 --name KafkaSparkHDFS /opt/bitnami/spark/jobs/spark_streaming_job.py. Error code is: 2.; 446)
[2024-10-31T21:27:03.001+0000] {local_task_job_runner.py:243} INFO - Task exited with return code 1
[2024-10-31T21:27:03.037+0000] {taskinstance.py:3503} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2024-10-31T21:27:03.046+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
